### PR TITLE
Add "add" and "drop" host config capability to DockerCreateContainer task

### LINKED ISSUE #1086: HostConfig Capability add and drop
### Expected Behavior

can add or drop capability list

### Current Behavior

can not add or drop capability list

### PR DESCRIPTION
**Pull request by denuno**
Add the capability (heh) to add or drop a List<String> of Capability to the HostConfig

closes #1086

### CODE REVIEW

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy [hunk 1]
```diff
@@ -429,6 +429,61 @@ class DockerWorkflowFunctionalTest extends AbstractGroovyDslFunctionalTest {
         build('workflow')
     }
 
+    def "Can build an image, create a container and add/drop capabilities"() {
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy:432]: This test should go into `DockerCreateContainerFunctionalTest`. I want to get rid of `DockerWorkflowFunctionalTest.groovy` long-term.

Comment (bmuschko) [reply src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy:432]: Another tests seems to be failing with your change on CI. Could you have a look? I am not 100% sure if they are related.

**Review by bmuschko — CHANGES_REQUESTED** [review]

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy (hunk 1 above)
Comment (denuno) [reply src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy]: Moved the test, and then that other one failing is just a change in output, I'll do a quick separate PR for that just cause

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy [hunk 2]
```diff
@@ -19,6 +19,8 @@ import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.TaskOutcome
 
+import static com.bmuschko.gradle.docker.TextUtils.escapeFilePath
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:22]: This is an unused import. Please remove.

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy [hunk 3]
```diff
@@ -353,6 +355,49 @@ class DockerCreateContainerFunctionalTest extends AbstractGroovyDslFunctionalTes
...
+        """
+
+        String containerInspect = '''
+            task inspectContainer(type: DockerInspectContainer) {
+                dependsOn startContainer
+                finalizedBy removeContainer
+                targetContainerId startContainer.getContainerId()
+
+                onNext { c ->
+                    if(c.hostConfig.capAdd.size() != 1) {
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:377]: Let's use the `empty` method here.

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy [hunk 4]
```diff
@@ -353,6 +355,49 @@ class DockerCreateContainerFunctionalTest extends AbstractGroovyDslFunctionalTes
...
+                        throw new GradleException("Invalid capAdd value!")
+                    }
+                    if(c.hostConfig.capAdd.first().toString() != 'NET_ADMIN') {
+                        throw new GradleException("Invalid capAdd value!")
+                    }
+                    if(c.hostConfig.capDrop.size() != 1) {
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:383]: Let's use the `empty` method here.

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy (hunk 3 above)
Comment (denuno) [reply src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:377]: this is an array, not a list, so there is no empty() method on it, unless I'm thinking of the wrong method

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy (hunk 4 above)
Comment (denuno) [reply src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:383]: same as above

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy (hunk 3 above)
Comment (bmuschko) [reply src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:377]: Got it. Forgot that we are checking a Docker Java type here.

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy (hunk 2 above)
Comment (bmuschko) [reply src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:22]: Can you remove this line?

**Comment by bmuschko** [comment]
Can you please merge the changes of https://github.com/bmuschko/gradle-docker-plugin/pull/1088 into here? It's in `master` now.

File: src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy [hunk 5]
```diff
@@ -19,6 +19,7 @@ import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.TaskOutcome
 
+
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy:22]: Can you remove the extra new line?

Comment (denuno) [reply src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy]: Sure!  Shoulda diffed it with master.  My IDE seems to be struggling with .editorconfig files, for whatever reason, at the moment, or I'da run a format on it.

Sidebar: I'm always both tempted and anti-tempted to go so far as to add Checkstyle, but it's not _too_ bad… :shrug: 

Thanks for getting at this so fast BTW!

**Pull request merged**